File: 백준/Bronze/17450. 과자 사기/과자 사기.py
import sys
input=lambda:sys.stdin.readline().rstrip()
res,ans=0,0
snack = ["S", "N", "U"]
for i in range(3):
    cost, weight = map(int, input().split())
    tcost, tweight = cost * 10, weight * 10
    money = tcost - 500 if tcost >= 5000 else tcost
    if (tweight / money) > res:
        res = tweight / money
        ans = i
print(snack[ans])

File: 백준/Bronze/17450. 과자 사기/과자 사기.c
#include <stdio.h>
#include <stdlib.h>
int main(void) {
    int cost, weight, idx = 0;
    double tcost, tweight, rlt = 0;
    char snacks[3][2] = {"S", "N", "U"};
    for (int i = 0; i < 3; i++) {
        scanf("%d %d", &cost, &weight);
        tcost = cost * 10 >= 5000 ? cost * 10 - 500 : cost * 10, tweight = weight * 10;
        if ((double)tweight / tcost > rlt) {
            rlt = tweight / tcost;
            idx = i;
        }
    }
    printf("%s", snacks[idx]);
    return 0;
}